cmd/gardener-extension-provider-aws/app/migrations.go

```diff
@@ -0,0 +1,53 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package app
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// TODO (georgibaltiev): Remove after the release of version 1.62.0
+func purgeMachineControllerManagerRBACResources(ctx context.Context, client client.Client) error {
```

Comment:
Could you add log entries when the runnable starts (at the place where we invoke this func), when we are about to delete a ClusterRole/ClusterRoleBinding and when the runnable (this func) ends?